core/dbt/include/global_project/macros/materializations/models/incremental/merge.sql

```diff
@@ -86,16 +86,16 @@
 {%- endmacro %}
 
 
-{% macro get_insert_overwrite_merge_sql(target, source, dest_columns, predicates, include_sql_header=false) -%}
-  {{ adapter.dispatch('get_insert_overwrite_merge_sql', 'dbt')(target, source, dest_columns, predicates, include_sql_header) }}
+{% macro get_insert_overwrite_merge_sql(target, source, dest_columns, predicates) -%}
+  {{ adapter.dispatch('get_insert_overwrite_merge_sql', 'dbt')(target, source, dest_columns, predicates) }}
 {%- endmacro %}
 
-{% macro default__get_insert_overwrite_merge_sql(target, source, dest_columns, predicates, include_sql_header) -%}
+{% macro default__get_insert_overwrite_merge_sql(target, source, dest_columns, predicates) -%}
     {%- set predicates = [] if predicates is none else [] + predicates -%}
     {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
     {%- set sql_header = config.get('sql_header', none) -%}
 
-    {{ sql_header if sql_header is not none and include_sql_header }}
+    {{ sql_header if sql_header is not none }}
```

Comment:
`include_sql_header` is indeed very very awkward. The reason we need it is that `dbt-bigquery` wants the `sql_header` in slightly different places for ["static"](https://github.com/dbt-labs/dbt-bigquery/blob/1b85fc50287ac9afd81fd92e715dffa5f0757379/dbt/include/bigquery/macros/materializations/incremental.sql#L49) versus ["dynamic"](https://github.com/dbt-labs/dbt-bigquery/blob/1b85fc50287ac9afd81fd92e715dffa5f0757379/dbt/include/bigquery/macros/materializations/incremental.sql#L83-L88) `insert_overwrite`, though I'm having a hard time remembering exactly why that is.

Reply:
Yup. Let me do a little digging, and try to follow up with an explanation of why it has to be that way :)

Reply:
@jtcohen6 Any more info on the need for the sql header in BigQuery?